### PR TITLE
Send Crud audit logs for integration with Fleek

### PR DESCRIPTION
**Pull request by dmolina79**
## Description
* Changes audit log to send all CRUD events
* adds ipfs information to request headers 
  IpfsContentTYpe = Bucket | Object
  IfpsHash

## Motivation and Context
We need integration in fleek to expose the bucket information to our platform.

Screen Shot 2020-04-21 at 2 28 43 PM
Screen Shot 2020-04-21 at 2 29 04 PM

### CODE REVIEW

File: cmd/logger/target/http/http.go [hunk 1]
```diff
@@ -20,12 +20,38 @@ import (
...
+var (
+	supportedApiEvents = []string{
+		"PutBucket",
+		"DeleteBucket",
+		"DeleteMultipleObjects",
+		"PutObject",
+		"CopyObject",
+		"DeleteObject",
+		"NewMultipartUpload",
+		// not sure about these below
```
Comment (jsonsivar) [inline cmd/logger/target/http/http.go:47]: For this maybe we can check which of the multipart operations actually updates the ipfs ledger for the root hash.  Feel like it should be the complete function so bucket hashes don't change in the middle if a multipart operation.

File: cmd/logger/target/http/http.go [hunk 2]
```diff
@@ -48,6 +48,13 @@ func (h *Target) startHTTPLogger() {
 	// from an internal channel.
 	go func() {
 		for entry := range h.logCh {
+			// TODO: add self call to info service to add hash to entry before sending payload
```
[the remark this answers is not in the document]
Comment (dmolina79) [reply cmd/logger/target/http/http.go]: actually that TODO should be removed, it was implemented in the PR. I missed that

File: cmd/logger/target/http/http.go [hunk 3]
```diff
@@ -94,9 +101,10 @@ func New(endpoint, userAgent, logKind string, transport *http.Transport) *Target
 
 // Send log message 'e' to http target.
 func (h *Target) Send(entry interface{}, errKind string) error {
-	if h.logKind != errKind && h.logKind != "ALL" {
+	// NOTE: commented out for Fleek to receive all events from Gateway
+/*	if h.logKind != errKind && h.logKind != "ALL" {
```
[the remark this answers is not in the document]
Comment (dmolina79) [reply cmd/logger/target/http/http.go]: all the levels are only for errors it seems, that check of errKind only looks for errors. So audit logs seems to be mostly for publishing errors out. But removing that piece we can expose all other crud events

File: cmd/logger/target/http/http.go (hunk 1 above)
Comment (dmolina79) [reply cmd/logger/target/http/http.go:47]: yeah, i can try that with some testing scripts as well

**Pull request merged**